docs/dev/developer-guides/transactions/fee-model.md

```diff
@@ -39,3 +36,67 @@ So the tips to make the most out of the zkSync fee system are the following:
 - **Reuse as many storage slots as possible.** Only the state diff is published on Ethereum.
 - **Users should share as many storage slots as possible.** If 100 users update a storage slot of your contract in a single block, the diff will be published only once. In the future, we will introduce reimbursement for the users, so that the costs for updating shared storage slots are split between the users.
 - **Reuse the contract code if possible.** On Ethereum, avoiding constructor parameters and putting them into constants reduces some of the gas costs upon contract deployment. On zkSync the opposite is true: deploying the same bytecode for contracts, while changing only constructor parameters can lead to substantial fee savings.
+
+
+## Gas estimation during transaction: For custom and paymaster accounts
+
+On Ethereum, there is a constant of 21000 gas that should cover all the intrinsic costs of processing a transaction: checking the signature and updating the nonce of the account. 
+
+On zkSync Era, it's quite different, because we support custom accounts that may take a slightly different (typically a bit higher) number of gas than EOAs. 
+We provide a convenient way for anyone to estimate the cost of a transaction regardless of the type of account. 
+### Changes in the `validateTransaction`
+
+Right now, the `validateTransaction` fails if the signature is not correct and returns `success=true` otherwise, while the `returndata` is discarded completely. We will migrate to a different approach, where the `validateTransaction` method to be considered successful must return some magic string. Any other `returndata` will be considered invalid and will be rejected by the system.  
```

Comment:
This was a part of the internal spec. Please remove this paragraph
```suggestion
It used to be the case for `validateTransaction` to fail if the signature is not correct and return `success=true` otherwise, while the `returndata` is discarded completely. We have migrated to a different approach, where the `validateTransaction` method to be considered successful must return a magic string. Any other `returndata` will be considered invalid and will be rejected by the system.  
```